### ISSUE TITLE
Cannot list messages or downloads

### ISSUE THREAD

**Issue by franburstall** [atlas-engineer/nyxt#2535]
**Describe the bug**

I attempt to `list-messages` and get an error:
```
<WARN> [10:33:54] Warning: Error on separate thread: too many colons in &quot;X&quot;

  Stream: #&lt;dynamic-extent STRING-INPUT-STREAM (unavailable) from &quot;/:X:/R/S...&quot;&gt;
```
and the same thing happens with `list-downloads`.

**Information**
- OS name+version:
```sh
debian-11
```
- Desktop environment / Window manager name+version: exwm
- How you installed Nyxt (Guix pack, package manager, build from source): source
- Information from `show-system-information`:
```
Nyxt version: 3-pre-release-1-341-gf49c768d
Renderer version: GI-GTK
Operating system kernel: Linux 5.10.0-17-amd64
Lisp implementation: SBCL 2.1.1 (Dynamic space size: 3221225472)
Features: (:QUICKLISP :NYXT-GI-GTK :NYXT-GTK :WEBKIT2 :WEBKIT2-2.36
 :WEBKIT2-CORS-ALLOWLIST :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING
 :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22
 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6
 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12
 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF
 :GLIB-2-30 :GLIB-2-32 :GLIB-2-34 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42
 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56
 :GLIB-2-58 :GLIB :NYXT-3.1 :NYXT-3 :CUSTOM-HASH-TABLE-NATIVE :SLYNK
 :PLUMP-UTF-32 :PARENSCRIPT :FSET-EXT-STRINGS :GLOBAL-VARS :DECLARE-TYPES
 :NAMED-READTABLES :SWANK :LPARALLEL :21BIT-CHARS :CLOSER-MOP :CL-PPCRE-UNICODE
 :CL-UNICODE :CL-PPCRE :CHUNGA :FLEXI-STREAMS :BORDEAUX-THREADS
 :LPARALLEL.WITH-CLTL2 :LPARALLEL.WITH-CAS :LPARALLEL.WITH-STEALING-SCHEDULER
 :SPLIT-SEQUENCE CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64
 CFFI-FEATURES:UNIX :CFFI CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP
 :FAST-IO-SV :FAST-IO :CL-JSON-CLOS :CL-JSON :SBCL-USES-SB-ROTATE-BYTE
 CHIPZ-SYSTEM:GRAY-STREAMS :THREAD-SUPPORT :ASDF3.3 :ASDF3.2 :ASDF3.1 :ASDF3
 :ASDF2 :ASDF :OS-UNIX :NON-BASE-CHARS-EXIST-P :ASDF-UNICODE :X86-64 :GENCGC
 :64-BIT :ANSI-CL :COMMON-LISP :ELF :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN
 :PACKAGE-LOCAL-NICKNAMES :SB-LDB :SB-PACKAGE-LOCKS :SB-THREAD :SB-UNICODE
 :SBCL :UNIX)

ASDF version: 3.3.1
ASDF registries: (NYXT-SOURCE-REGISTRY ENVIRONMENT-SOURCE-REGISTRY USER-SOURCE-REGISTRY
                      USER-SOURCE-REGISTRY-DIRECTORY
                      DEFAULT-USER-SOURCE-REGISTRY SYSTEM-SOURCE-REGISTRY
                      SYSTEM-SOURCE-REGISTRY-DIRECTORY
                      DEFAULT-SYSTEM-SOURCE-REGISTRY)
Critical dependencies: (/home/fran/common-lisp/nyxt/_build/cl-cffi-gtk/gtk/cl-cffi-gtk.asd
 /home/fran/common-lisp/nyxt/_build/cl-gobject-introspection/cl-gobject-introspection.asd
 /home/fran/common-lisp/nyxt/_build/cl-webkit/webkit2/cl-webkit2.asd)


```

**Comment by aadcg** [comment]
I'm not able to reproduce.

**Comment by franburstall** [comment]
Not sure where to go with this then.  I get this error with many commands that fire an internal page: `new`, `manual`, `list-history` and more.  This is with `nyxt -I` so my config should not play a role.  But my history does get loaded: is there a way to fire nyxt without that?

**Comment by franburstall** [comment]
OK---it was my history: moving default.lisp from its usual place enabled nyxt to start up normally without all these errors.

However, this prompts a new question: how can a history file (which was less than a week old) poison nyxt so thoroughly?  

I attach the bad history file in case it is helpful.
[bad-default.txt](https://github.com/atlas-engineer/nyxt/files/9482513/bad-default.txt)

**Comment by Ambrevar** [comment]
We need to look into this.

In any case, bad history files have broken Nyxt for too long.  We need to implement 2 measures:

1. Fix the startup fallback.  I already implemented a fall back measure for when the history does not load, but apparently it's not enough.
2. Add a CLI option to skip loading history.  Note that it's already doable using the `--with-file` option, but it's not straightforward enough for something that is that needed.  Better: add a single fool-proof startup flag that skips auto-config, config and all kinds of file loading.

**Comment by aartaka** [comment]
To add to the history problems: if the first buffer to restore was the one with some Internet link, loading it with no Internet connection will hang Nyxt.

Same happens with faulty internal pages—if those error the moment session restores, the whole browser stays gray.

**Comment by hendursaga** [comment]
I wonder if a [SQLite backend](https://github.com/atlas-engineer/nyxt/issues/2435#issuecomment-1207442591) would have less troubles... :)

**Comment by Ambrevar** [comment]
It's unrelated: SQLite deals with persistence, here the problem is the startup logic.

**Comment by aartaka** [comment]
Does eebf1f8d72cc2b394140afc9c8fed58802f8c461 fix it, @franburstall?

**Comment by franburstall** [comment]
> Does [eebf1f8](https://github.com/atlas-engineer/nyxt/commit/eebf1f8d72cc2b394140afc9c8fed58802f8c461) fix it, @franburstall?

Yes!  nyxt loads my "bad" history file and functions normally.

**Comment by franburstall** [comment]
> 2. Add a CLI option to skip loading history.  Note that it's already doable using the `--with-file` option,

Question: how to use `--with-files` to skip history loading?  Neither of
```
nyxt --with-files session=
```
or
```
nyxt --with-files history=
```
work for me.

**Comment by franburstall** [comment]
> > Does [eebf1f8](https://github.com/atlas-engineer/nyxt/commit/eebf1f8d72cc2b394140afc9c8fed58802f8c461) fix it, @franburstall?
> 
> Yes! nyxt loads my "bad" history file and functions normally.

Sorry: I spoke too soon.  With the bad history file, many things work but `list-buffers` errors out with:
```
Warning: Error while processing the "nyxt:" URL: The value 1125 is not of type (OR (VECTOR CHARACTER) BASE-STRING SYMBOL CHARACTER) when binding SB-IMPL::STRING1
```

**Comment by Ambrevar** [comment]
> Question: how to use `--with-files` to skip history loading?

Oops!  Oversight on my side, should be fixed with 10bc9647f9a9536b9bd0eb830d5d392b2474564e.

Syntax is `nyxt --with-path history=`.

**Comment by Ambrevar** [comment]
> but `list-buffers` errors out with:

Fixed with 9f74e8cd2ac1ad6e683a19bd3d31223691e66f0b.

**Comment by Ambrevar** [comment]
See also #2575.
